File: src/main/java/org/railwaystations/rsapi/core/model/ProblemReport.java
package org.railwaystations.rsapi.core.model;

import lombok.Builder;
import lombok.Value;

@Value
@Builder
public class ProblemReport {

    String countryCode;

    String stationId;

    String title;

    Long photoId;

    ProblemReportType type;

    String comment;

    Coordinates coordinates;

}

File: service/src/main/kotlin/org/railwaystations/rsapi/core/model/ProblemReport.kt
package org.railwaystations.rsapi.core.model

data class ProblemReport(
    val countryCode: String,
    val stationId: String,
    val title: String? = null,
    val photoId: Long? = null,
    val type: ProblemReportType,
    val comment: String? = null,
    val coordinates: Coordinates? = null,
)
